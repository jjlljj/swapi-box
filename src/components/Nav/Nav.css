.nav-menu {
  text-align: center;
  margin: 20px 0 10px;
}

.nav-menu div {
  display: inline-block;
  background: transparent;
}

.nav-menu a {
  display: block;
  font-size: 16px;
  font-weight: 400;
  height:100%;
  padding: 6px;
  width: 100px;
}

.active {
  background: #feda4a;
  color: black;
}

@media screen and (max-width: 450px) {
  

  .nav-menu div {
    display: block;
  }

  .nav-menu a {
    padding: 10px 0;
    width: 100%;
  }
}
