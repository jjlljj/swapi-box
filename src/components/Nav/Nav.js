import React from 'react';
import './Nav.css';
import { number } from 'prop-types';
import { NavLink } from 'react-router-dom';

const Nav = ({ favLength }) => {
  const favNum = favLength || 0;

  return (
    <nav className="nav-menu">
      <div>
        <NavLink 
          to="/people"
          activeClassName="active" >
          People
        </NavLink>
      </div>
      <div>
        <NavLink 
          to="/planets"
          activeClassName="active" >
          Planets
        </NavLink>
      </div>
      <div>
        <NavLink 
          to="/vehicles"
          activeClassName="active" >
          Vehicles
        </NavLink>
      </div>
      <div>
        <NavLink 
          to="/favorites"
          activeClassName="active">
          Favorites
          <span className='fav-num'>{favNum}</span>
        </NavLink>
      </div>
    </nav>
  );
};

Nav.propTypes = {
  favLength: number
};

export default Nav;
