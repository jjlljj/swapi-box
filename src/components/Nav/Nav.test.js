/* eslint-disable */

import React from 'react';
import ReactDom from 'react-dom';
import { shallow } from 'enzyme';
import Nav from './Nav';

describe('Nav', () => {

  it('should exist', () => {
    const renderedComponent = shallow(<Nav />);

    expect(renderedComponent).toBeDefined();
  })

})
