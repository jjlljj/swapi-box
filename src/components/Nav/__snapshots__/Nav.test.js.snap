// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should exist 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/people"
          >
                    People
          </NavLink>
</div>,
        <div>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/planets"
          >
                    Planets
          </NavLink>
</div>,
        <div>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/vehicles"
          >
                    Vehicles
          </NavLink>
</div>,
        <div>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/favorites"
          >
                    Favorites
                    <span
                              className="fav-num"
                    />
          </NavLink>
</div>,
      ],
      "className": "nav-menu",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/people"
>
            People
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "People",
            "to": "/people",
          },
          "ref": null,
          "rendered": "People",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/planets"
>
            Planets
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Planets",
            "to": "/planets",
          },
          "ref": null,
          "rendered": "Planets",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/vehicles"
>
            Vehicles
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Vehicles",
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/favorites"
>
            Favorites
            <span
                        className="fav-num"
            />
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": Array [
              "Favorites",
              <span
                className="fav-num"
/>,
            ],
            "to": "/favorites",
          },
          "ref": null,
          "rendered": Array [
            "Favorites",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0,
                "className": "fav-num",
              },
              "ref": null,
              "rendered": 0,
              "type": "span",
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/people"
            >
                        People
            </NavLink>
</div>,
          <div>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/planets"
            >
                        Planets
            </NavLink>
</div>,
          <div>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/vehicles"
            >
                        Vehicles
            </NavLink>
</div>,
          <div>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/favorites"
            >
                        Favorites
                        <span
                                    className="fav-num"
                        />
            </NavLink>
</div>,
        ],
        "className": "nav-menu",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/people"
>
              People
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "People",
              "to": "/people",
            },
            "ref": null,
            "rendered": "People",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/planets"
>
              Planets
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Planets",
              "to": "/planets",
            },
            "ref": null,
            "rendered": "Planets",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/vehicles"
>
              Vehicles
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Vehicles",
              "to": "/vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/favorites"
>
              Favorites
              <span
                            className="fav-num"
              />
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": Array [
                "Favorites",
                <span
                  className="fav-num"
/>,
              ],
              "to": "/favorites",
            },
            "ref": null,
            "rendered": Array [
              "Favorites",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "fav-num",
                },
                "ref": null,
                "rendered": 0,
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
